conf/context.go

```diff
@@ -0,0 +1,181 @@
+package conf
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+)
+
+type audiusContextKey string
+
+const (
+	ConfigVersion = "0.1"
+	ContextKey    = audiusContextKey("audius_cobra_context")
+)
+
+func ReadOrCreateContextConfig() (*ContextConfig, error) {
+	confDir, err := getConfigBaseDir()
+	if err != nil {
+		return nil, err
+	}
+
+	execConfFilePath := filepath.Join(confDir, "audius")
+	err = os.MkdirAll(filepath.Join(confDir, "contexts"), os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+	if _, err := os.Stat(execConfFilePath); os.IsNotExist(err) {
+		fmt.Println("No existing config found at ~/.audius, creating new.")
+		createExecutionConfig(execConfFilePath)
+	}
+
+	var execConf ExecutionConfig
+	if err = readExecutionConfig(&execConf); err != nil {
+		return nil, err
+	}
+
+	contextDir, err := getContextBaseDir()
+	if err != nil {
+		return nil, err
+	}
+	contextFilePath := filepath.Join(contextDir, execConf.CurrentContext)
+	if _, err = os.Stat(contextFilePath); os.IsNotExist(err) {
+		fmt.Printf("Context '%s' not found, creating default.\n", execConf.CurrentContext)
+		createDefaultContext(contextFilePath)
+		err = SetContext("default")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var ctx ContextConfig
+	err = readConfigFromFile(contextFilePath, &ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ctx, nil
```

Comment:
Is this just as bad in golang land as it is in C++ land? Should the function instead require the caller to pass in a pointer to their own ContextConfig?

Reply:
you mean have a function signature like
```
func ReadContextConfig(ctx *ContextConfig) (*ContextConfig, error) {
    // modify passed in ctx here and return?

}
```

Reply:
similar to how the json marshal package works?

Reply:
Looks like we can do it either way https://stackoverflow.com/questions/46987513/handling-dangling-pointers-in-go